package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -552,16 +561,57 @@ def _read_next_timestep(self):
 
         indices = np.zeros(self.n_atoms, dtype=int)
 
-        f.readline()  # ITEM ATOMS etc
-        for i in range(self.n_atoms):
-            idx, _, xs, ys, zs = f.readline().split()
+        atomline = f.readline()  # ITEM ATOMS etc
+        attrs = atomline.split()[2:]  # attributes on coordinate line
+        col_ids = {}  # column index of each attribute
+        for i, attr in enumerate(attrs):
+            col_ids[attr] = i
+
+        # check for ids and what type of coordinate convention
+        ids = "id" in col_ids.keys()
+        # keys = convention values = col_ids or False if not present
+        coord_dict = {"unscaled": False, "scaled": False, "unwrapped": False,
+                      "scaled_unwrapped":False}
+        if "x" and "y" and "z" in col_ids.keys():  # unscaled
+            coord_dict["unscaled"] = [col_ids["x"], col_ids["y"],
+                                      col_ids["z"]]
+        if "xs" and "ys" and "zs" in col_ids.keys():  # scaled
+            coord_dict["scaled"] = [col_ids["xs"], col_ids["ys"],
+                                    col_ids["zs"]]
+        if "xu" and "yu" and "zu" in col_ids.keys():  # unwrapped
+            coord_dict["unwrapped"] = [col_ids["xu"], col_ids["yu"],
+                                       col_ids["zu"]]
+        if "xsu" and "ysu" and "zsu" in col_ids.keys():  # scaled unwrapped
+            coord_dict["scaled_unwrapped"] = [col_ids["xsu"], col_ids["ysu"],
+                                              col_ids["zsu"]]
+
+        # this should only trigger on first read of "ATOM" card, after which it
+        # is fixed to the guessed value.
+        if self.coordinate_convention == "guess":
+            for convention in self._conventions[1:]:
+                if convention in coord_dict.keys() and coord_dict[convention]:
+                    coord_cols = coord_dict[convention]
+                    self.coordinate_convention = convention
+
+        if not coord_dict[self.coordinate_convention]:
+            raise ValueError(f"no coordinate information of type"
+                             f"{self.coordinate_convention} in frame")
+
+        else:
+            coord_cols = coord_dict[self.coordinate_convention]
 
-            indices[i] = idx
-            ts.positions[i] = xs, ys, zs
+        for i in range(self.n_atoms):
+            fields = f.readline().split()
+            if ids:
+                indices[i] = fields[col_ids["id"]]
+            x, y, z = [fields[i] for i in coord_cols]
+            ts.positions[i] = x, y, z
 
         order = np.argsort(indices)
         ts.positions = ts.positions[order]
-        # by default coordinates are given in scaled format, undo that
-        ts.positions = distances.transform_StoR(ts.positions, ts.dimensions)
+        if self.coordinate_convention == "scaled":
```

Comment:
Or scaled unwrapped

Reply:
Does scaled_unwrapped use the same S->R? I wasn't sure so i left it out.

Reply:
Yeah I'd want to test to make sure but I'm assuming scaled_unwrapped can be 1.5 to mean it's 0.5 * boxlength + 1 boxlength

Reply:
Correct in that it uses same S->R.